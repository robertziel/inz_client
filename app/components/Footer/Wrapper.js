import styled from 'styled-components';

const Wrapper = styled.footer`
  position: fixed;
  bottom: 0px;
  width: 100%;
  background-color: #ffffff;
  border-top: rgba(0, 0, 0, 0.02) solid 1px;
  padding: 10px;

  .message {
    text-transform: uppercase;

    a {
      color: #00a9ff;
      text-decoration: none;
      font-weight: 600;
    }
  }
`;

export default Wrapper;
