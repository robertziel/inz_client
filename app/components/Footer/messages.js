import { defineMessages } from 'react-intl';

export const scope = 'app.components.Footer';

const messages = defineMessages({
  authorMessage: {
    id: `${scope}.author.message`,
    defaultMessage: `
      Made by {author}.
    `,
  },
});

export default messages;
