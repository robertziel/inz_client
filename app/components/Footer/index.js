import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import Wrapper from './Wrapper';
import messages from './messages';

export default function Footer() {
  return (
    <Wrapper>
      <section>
        <div className='message'>
          <FormattedMessage
            {...messages.authorMessage}
            values={{
              author: <a href='https://robertz.co' target='_blank'>Robert Zieli≈Ñski</a>,
            }}
          />
        </div>
      </section>
    </Wrapper>
  );
}
