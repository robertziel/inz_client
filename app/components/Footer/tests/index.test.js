import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Footer from '../index';

const renderer = new ShallowRenderer();

describe('<Footer />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<Footer />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
