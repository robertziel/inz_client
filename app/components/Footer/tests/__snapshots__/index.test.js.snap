// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
"<footer class=\\"Wrapper-wcfdfo-0 bzeFoT\\"><section><div class=\\"message\\"><span>
      Made by <a href=\\"https://robertz.co\\" target=\\"_blank\\">Robert Zieli≈Ñski</a>.
    </span></div></section></footer>"
`;
